app/scripts/lib/transaction/metrics.ts

```diff
@@ -800,6 +800,14 @@ async function buildEventFragmentProperties({
     transactionMetricsRequest.getTokenStandardAndDetails,
   );
 
+  let contractMethodName;
+  if (transactionMeta.txParams.data) {
```

Comment:
Minor, but given the size of this function already, could this go in a small:

`getContractMethodName(request, transactionMeta)`

Reply:
I would avoid another function for this code its only 6-7 lines.